File: fileshare-mainprocess/fileshare-mainprocess/HelloSenderThread.cpp
#include "stdafx.h"
#include "HelloSenderThread.h"
#include <Windows.h>

HelloSenderThread::HelloSenderThread()
{
}


HelloSenderThread::~HelloSenderThread()
{
}

void HelloSenderThread::operator() () {
	while (1) {
		//string ipadd = NetConf::getLocalipaddr() == "none" ? "noip:RAND:" + rand() : NetConf::getLocalipaddr();
		HelloNetworkModule& network = HelloNetworkModule::getInstance();
		network.send(KeepalivePacket(network.getPeer().id,network.getPeer().name,network.getPeer().ipaddress));
		Sleep(10000);
	}

}

File: fileshare-csharp/HelloProtocolPorting/HelloProtocol/HelloSenderThread.cs
using System.Threading;

namespace HelloProtocol {

    internal class HelloSenderThread : ExecutableThread {

        public HelloSenderThread() {

        }

        protected override void execute() {
            HelloNetworkModule network = HelloNetworkModule.Instance;
            while(true) {
                network.send(new KeepalivePacket(Settings.Instance.LocalPeer.Id,
                                                 Settings.Instance.LocalPeer.Name,
                                                 Settings.Instance.LocalPeer.Ipaddress));
                Thread.Sleep(Settings.Instance.HELLO_INTERVAL);
            }
        }
    }
}